ethereum/blockwatch/client.go

```diff
@@ -54,6 +61,8 @@ type GetBlockByNumberResponse struct {
 // HeaderByNumber fetches a block header by its number. If no `number` is supplied, it will return the latest
 // block header. If no block exists with this number it will return a `ethereum.NotFound` error.
 func (rc *RpcClient) HeaderByNumber(number *big.Int) (*miniheader.MiniHeader, error) {
+	rc.rateLimiter.Wait(context.Background())
```

Comment:
If I understand correctly, using `context.Background()` here means that we will wait forever until it is possible to send the request. I don't think this is wise. At the very least, we should be inheriting from some parent context so that if we are in the process of shutting down Mesh, we don't keep waiting forever here. This is going to be a requirement for implementing graceful shutdowns.

On top of that, I'm worried that this can lead to a situation where we are waiting for new requests before the existing requests have a chance to be sent. In many cases, I think we _do_ want to give up after too much time has passed. For example if we are trying to get block headers for a particular block range, after too much time has passed that request will be outdated. Or if we are trying to validate an order, the order could expire or become unfillable, making the request irrelevant.

I think we should both inherit from a parent context and use `context.WithTimeout` to add an additional timeout.

Reply:
Agreed. We are now passing down a context here. There are still other places were we don't do this, so I've added a note to https://github.com/0xProject/0x-mesh/issues/161 that we should re-visit all uses of `context.Background()` to enable graceful shutdowns.